src/unix/fs.c

```diff
@@ -217,6 +217,30 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  static int no_cloexec_support;
+  int fd;
+
+#ifdef O_CLOEXEC
+  no_cloexec_support = 1;
```

Comment:
See what we do in the fallback path in `uv__fs_open`. You need to take the cloexec lock and use `uv__cloexec`. Also close the fd in case of error.